.github/workflows/sbom.yml

```diff
@@ -0,0 +1,52 @@
+name: SBOM generation
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  sbom_gen:
+    name: Generate SBOM
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    continue-on-error: true
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+          cache: 'pip'
+          cache-dependency-path: '**/requirements.txt'
+      - name: Get date
+        id: get-date
+        run: |
+          echo "::set-output name=date::$(/bin/date -u "+%Y%m%d")"
+      - name: Get cached database
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/cve-bin-tool
+          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+      - name: Install dependencies and cve-bin-tool
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade setuptools
+          python -m pip install --upgrade wheel
+          python -m pip install --upgrade pytest
+          python -m pip install --upgrade sbom4python
+          pip install . -r doc/requirements.txt
+      - name: Generate SBOM for cve-bin-tool
+        run: |
+          sbom4python --module cve-bin-tool --output cve-bin-tool.spdx
+      - name: Compare SBOM for cve-bin-tool
+        # This would fail due to time/date of SBOM generation in SBOM header
+        # Therefore ignore first 10 lines of file in comparison which is SBOM header
+        run: |
+          /bin/tail -n +10 sbom/cve-bin-tool.spdx > orig
+          /bin/tail -n +10 cve-bin-tool.spdx > new
+          /bin/diff -b orig new
```

Comment:
```suggestion
          /bin/diff -b orig new || true
```

Reply:
Lazy solution to our diff fail problem.  The internet tells me that `||:` would be more bash-y but this is probably more readable.  

Unfortunately, if this works it might break the `cat` check below so you might have to do some fancy footwork to avoid that.  Possibly something like piping the diff output to a file then running a size check instead?

Reply:
@terriko A size check for the file might not work if we just have a small change in a version number e.g. 0.1 => 0.2 in a component. I wonder if doing a checksum comparison might be better as this would detect any changes in the content. 

I think we need to only run this action on request and not on every PR. Maybe every time we do a release or maybe every month?

Reply:
Yeah, if we do is just once a month out of band with PRs then it's fine if it fails, so we should probably just do that.  I can do it manually per release.  I should probably make a published release checklist and add it to that.